File: 812-rotate-string/rotate-string.cpp
class Solution {
public:
    bool rotateString(string s, string goal) {
       int count=0;
       int n=s.size();
       while(count<n)
       {
        if(s==goal)
        {
            return true;
        }
        char x=s.back();
        s.pop_back();
        s=x+s;
        ++count;
       }
       return false;

    }
};

File: 812-rotate-string/rotate-string.java
class Solution {
    public boolean rotateString(String s, String goal) {
          if (s.length() != goal.length()) return false;
        
        for (int i = 0; i < s.length(); i++) {
            // Correctly rotate s by taking the suffix and adding the prefix
            String rotated = s.substring(i) + s.substring(0, i);
            if (rotated.equals(goal)) {
                return true;
            }
        }
        return false;
    }
}